src/DataProvider/SubcollectionDataProviderInterface.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\DataProvider;
+
+use ApiPlatform\Core\Exception\ResourceClassNotSupportedException;
+
+/**
+ * Retrieves subcollections from a persistence layer.
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+interface SubcollectionDataProviderInterface
```

Comment:
We might but the dataprovides should understand that we "need the collection belonging to an item", and that's very different from both of the current implementations IMO.